File: jscomp/test/test_require.ml


let () =
  match [%node require] with
  | None ->   ()
  | Some u ->               
    match [%node _module], Js.Undefined.toOption u##main with
    | Some x, Some y when x == y ->
      Js.log "is main"
    | Some _, Some _
    | None, Some _ 
    | None, None  
    | Some _, None -> 
      Js.log "not main"

File: jscomp/test/test_require.js
'use strict';


var u = typeof require === "undefined" ? undefined : require;

if (u !== undefined) {
  var match = typeof module === "undefined" ? undefined : module;
  var match$1 = u.main;
  if (match !== undefined && match$1 !== undefined && match === match$1) {
    console.log("is main");
  } else {
    console.log("not main");
  }
}

/* u Not a pure module */
